lib/benchee/conversion/memory.ex

```diff
@@ -48,125 +48,37 @@ defmodule Benchee.Conversion.Memory do
-  end
-  def convert({value, %Unit{name: :megabyte}}, kilobyte_unit = %Unit{name: :kilobyte}) do
-    converted_value = multiply_by_thousands_of_bytes(value, 1)
-    {converted_value, kilobyte_unit}
-  end
-  def convert({value, %Unit{name: :megabyte}}, gigabyte_unit = %Unit{name: :gigabyte}) do
-    converted_value = divide_by_thousands_of_bytes(value, 1)
-    {converted_value, gigabyte_unit}
-  end
-  def convert({value, %Unit{name: :megabyte}}, terabyte_unit = %Unit{name: :terabyte}) do
-    converted_value = divide_by_thousands_of_bytes(value, 2)
-    {converted_value, terabyte_unit}
-  end
-
-  # Gigabytes conversions
-  def convert({value, %Unit{name: :gigabyte}}, byte_unit = %Unit{name: :byte}) do
-    converted_value = multiply_by_thousands_of_bytes(value, 3)
-    {converted_value, byte_unit}
-  end
-  def convert({value, %Unit{name: :gigabyte}}, kilobyte_unit = %Unit{name: :kilobyte}) do
-    converted_value = multiply_by_thousands_of_bytes(value, 2)
-    {converted_value, kilobyte_unit}
-  end
-  def convert({value, %Unit{name: :gigabyte}}, megabyte_unit = %Unit{name: :megabyte}) do
-    converted_value = multiply_by_thousands_of_bytes(value, 1)
-    {converted_value, megabyte_unit}
-  end
-  def convert({value, %Unit{name: :gigabyte}}, terabyte_unit = %Unit{name: :terabyte}) do
-    converted_value = divide_by_thousands_of_bytes(value, 1)
-    {converted_value, terabyte_unit}
-  end
-
-  # Terabytes conversions
-  def convert({value, %Unit{name: :terabyte}}, byte_unit = %Unit{name: :byte}) do
-    converted_value = multiply_by_thousands_of_bytes(value, 4)
-    {converted_value, byte_unit}
-  end
-  def convert({value, %Unit{name: :terabyte}}, kilobyte_unit = %Unit{name: :kilobyte}) do
-    converted_value = multiply_by_thousands_of_bytes(value, 3)
-    {converted_value, kilobyte_unit}
-  end
-  def convert({value, %Unit{name: :terabyte}}, megabyte_unit = %Unit{name: :megabyte}) do
-    converted_value = multiply_by_thousands_of_bytes(value, 2)
-    {converted_value, megabyte_unit}
-  end
-  def convert({value, %Unit{name: :terabyte}}, gigabyte_unit = %Unit{name: :gigabyte}) do
-    converted_value = multiply_by_thousands_of_bytes(value, 1)
-    {converted_value, gigabyte_unit}
-  end
-
-  defp multiply_by_thousands_of_bytes(value, amount) when amount > 0 do
-    value * @bytes_per_kilobyte * amount
+  @spec convert(value_with_unit, unit) :: value_with_unit
+  def convert({value, %Unit{magnitude: current_magnitude}}, desired_unit = %Unit{magnitude: desired_magnitude}) do
+    multiplier = current_magnitude / desired_magnitude
+    {value * multiplier, desired_unit}
   end
 
-  defp divide_by_thousands_of_bytes(value, 0) do
-    value
-  end
-  defp divide_by_thousands_of_bytes(value, amount) when amount > 0 do
-    divide_by_thousands_of_bytes(value / @bytes_per_kilobyte, amount - 1)
+  def convert({value, current_unit}, desired_unit) when current_unit == desired_unit do
+    {value, current_unit}
   end
```

Comment:
I don't think we need this special case (also it will never get hit as pattern matches are evaluated top to bottm) - in this case both magnitues are the same, dividing them will be 1 and so everything should be fine :)

Reply:
All fair points! Will change accordingly